.changeset/chatty-ravens-collect.md

```diff
@@ -0,0 +1,29 @@
+---
+"@vivjs/constants": minor
+"@vivjs/extensions": minor
+"@vivjs/layers": minor
+"@vivjs/loaders": minor
+"@hms-dbmi/viv": minor
+"@vivjs/types": minor
+"@vivjs/viewers": minor
+"@vivjs/views": minor
+---
+
+This release potentially includes backward-incompatible changes. To avoid automatically adopting such releases, ensure you are either pinning the exact version of @hms-dbmi/viv in your package.json file or using a version range syntax that only accepts patch updates, such as ~0.16.1. Refer to npm's semver documentation for details.
+
+    Migrate to deck.gl v9
+
+This update modifies our code internally to support deck.gl v9, dropping compatibility with deck.gl v8.
+
+Impact:
+
+We have not made any public API changes from this except the version bump, and what these new versions require "internally" to your GL related code:
```

Comment:
I mean if someone was using the ability to make custom extensions, that would be a breaking change, right? I just don't think we need to shy around the language about changes.

The TL;DR for me is that if you are relying deeply on Viv's layers and internals, this might be harder migration but hopefully Deck.gl 9 is on your radar and something you're working towards and understand the changes (and will welcome these).

If you are using the react components/data lodaers... much less so as you probably aren't working with deck directly.

Reply:
Like deprecating WebGL1 is a totally a public breaking change, even if its something that is "internal".

Reply:
I agree—I would be distinct about "breaking" and the relationship with public APIs (i.e., breaking changes doesn't mean we've changed interfaces). 

We should clearly state that this upgrade is "breaking" due to the underlying changes required to support deck.gl 9. While Viv's public APIs haven’t changed, these dependency updates can affect users and require migration effort.

> We haven’t changed Viv's public API, but the upgrade to deck.gl 9.0 is considered breaking due to changes in its dependencies, which may require updates to WebGL-related code (e.g., shaders, injections, constants). Here are potential issues users may face: